python/lsst/ts/tunablelaser/mock_server.py

```diff
@@ -1,45 +1,51 @@
 import logging
-import queue
-import serial
 import inspect
+import asyncio
 
+from lsst.ts import tcpip
 
-class MockSerial(serial.Serial):
-    def __init__(
-        self,
-        port,
-        baudrate=19200,
-        bytesize=serial.EIGHTBITS,
-        parity=serial.PARITY_NONE,
-        stopbits=serial.STOPBITS_ONE,
-        timeout=3,
-        xonxoff=False,
-        rtscts=False,
-        write_timeout=None,
-        dsrdtr=False,
-        inter_byte_timeout=None,
-        exclusive=None,
-    ):
-        super().__init__(port=port)
-        self.log = logging.getLogger(__name__)
 
+class MockServer(tcpip.OneClientServer):
+    """Simulates the TunableLaser."""
+
+    def __init__(self) -> None:
         self.device = MockNT900()
-        self.message_queue = queue.Queue()
+        self.log = logging.getLogger(__name__)
+        self.read_loop_task = asyncio.Future()
+        self.lock = asyncio.Lock()
+        super().__init__(
+            name="TunableLaser Mock Server",
+            host=tcpip.LOCAL_HOST,
+            port=9999,
```

Comment:
What is 9999? If this is the real port for the laser then please use a constant (e.g. LASER_PORT). If this is a random possibly free port then please use 0 for that purpose -- to get a port that is guaranteed to be available.

Unit tests should always use port 0 to create the server, and then examine the chosen port in order to create the client. That avoids conflicts and makes the tests far more robust.